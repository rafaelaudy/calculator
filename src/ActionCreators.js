export const NUMBER_CLICK_TYPE = "NUMBER_CLICK";
export const numberClick = number => ({
  type: NUMBER_CLICK_TYPE,
  payload: { value: number + "" }
});

export const SUM_CLICK_TYPE = "SUM_CLICK";
export const sumClick = () => ({
  type: SUM_CLICK_TYPE
});

export const SUBTRACTION_CLICK = "SUBTRACTION_CLICK";
export const subtractionClick = () => ({
  type: SUBTRACTION_CLICK
});

export const MULTIPLICATION_CLICK = "MULTIPLICATION_CLICK";
export const multiplicationClick = () => ({
  type: MULTIPLICATION_CLICK
});

export const DIVISION_CLICK = "DIVISION_CLICK";
export const divisionClick = () => ({
  type: DIVISION_CLICK
});
