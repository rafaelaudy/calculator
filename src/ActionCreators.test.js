import { numberClick } from "./ActionCreators";

it("numberClick action creator exports the correct format", () => {
  expect(numberClick(1)).toMatchSnapshot();
});
