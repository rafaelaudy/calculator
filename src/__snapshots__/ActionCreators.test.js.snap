// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateClick action creator exports the correct format 1`] = `
Object {
  "type": "CALCULATE_CLICK",
}
`;

exports[`divisionClick action creator exports the correct format 1`] = `
Object {
  "type": "DIVISION_CLICK",
}
`;

exports[`multiplicationClick action creator exports the correct format 1`] = `
Object {
  "type": "MULTIPLICATION_CLICK",
}
`;

exports[`numberClick action creator exports the correct format 1`] = `
Object {
  "payload": Object {
    "input": "1",
  },
  "type": "NUMBER_CLICK",
}
`;

exports[`subtractionClick action creator exports the correct format 1`] = `
Object {
  "type": "SUBTRACTION_CLICK",
}
`;

exports[`sumClick action creator exports the correct format 1`] = `
Object {
  "type": "SUM_CLICK",
}
`;
